mono/mini/Makefile.am.in

```diff
@@ -34,6 +34,52 @@ sgen_libs = \
 	$(monodir)/mono/utils/libmonoutils.la \
 	$(glib_libs)
 
+if ENABLE_LLVM
+# We need to define
+#    LLVM_CFLAGS
+#    LLVM_CXXFLAGS
+#    LLVM_LIBS
+#    LLVM_LDFLAGS
+
+if MONO_INTERNAL_LLVM 
+llvm_config=$(monodir)/llvm/usr/bin/llvm-config
+else
+llvm_config=$(EXTERNAL_LLVM_CONFIG)
+endif
+llvm_system_libs=$(shell $(llvm_config) --system-libs 2>/dev/null | grep -- -)
+LLVM_VERSION=$(shell $(llvm_config) --version)
+
+llvm_api_version_from_config=$(shell $(llvm_config) --mono-api-version 2>/dev/null)
+llvm_api_version=$(shell if [  -z "$(llvm_api_version_from_config)" ]; then echo "0"; else echo "$(llvm_api_version_from_config)"; fi)
+
+llvm_cflags_from_config=$(shell $(llvm_config) --cflags | sed -e 's/-O2//g' | sed -e 's/-O0//g' | sed -e 's/-fomit-frame-pointer//g' | sed -e 's/-fPIC//g')
+llvm_cxxflags_from_config=$(shell $(llvm_config) --cxxflags | sed -e 's/-O2//g' | sed -e 's/-O0//g' | sed -e 's/-fomit-frame-pointer//g' | sed -e 's/-fPIC//g')
+LLVM_CFLAGS_INTERNAL=$(llvm_cflags_from_config) -DLLVM_API_VERSION=$(llvm_api_version)
+LLVM_CXXFLAGS_INTERNAL=$(llvm_cxxflags_from_config) -fno-rtti -fexceptions -DLLVM_API_VERSION=$(llvm_api_version)
+
+llvm_common_libs=$(shell $(llvm_config) --libs analysis core bitwriter $(LLVM_CODEGEN_LIBS))
+# The one that doesn't work will be null, so just concatenate them
+llvm_libs_for_old=$(shell $(llvm_config) --libs mcjit jit)
+llvm_libs_for_new=$(shell $(llvm_config) --libs orcjit)
```

Comment:
`llvm_libs_for_new:=$(shell $(llvm_config) --libs orcjit)`

Why don't we just merge `llvm_libs_for_old` and `llvm_libs_for_new`?

Reply:
We can use the two separate shell commands inline without making new variables, but the line gets rather wide.

Reply:
My question was more, why can't we have `$(shell $(llvm_config) --libs mcjit jit orcjit)`?

Reply:
When we are using an older llvm, `jit` is a valid lib name. When we are using a newer one, mcjit is a valid lib name. So at any given time, one of these commands are going to error out and return "". If we had both args, we would get "" as the jit lib line on every version, because no version defines both jit and mcjit as valid lib names.

Reply:
It will probably fail the command so there will be no output.

Reply:
This was my trick to get around not being able to do complex comparisons to find out which one to run. We run both, and one of them will work and the other one will gracefully fail.

Reply:
Ok 👍